src/sqlancer/mysql/oracle/MySQLDQEOracle.java

```diff
@@ -0,0 +1,502 @@
+package sqlancer.mysql.oracle;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.beust.jcommander.Strings;
+
+import sqlancer.Randomly;
+import sqlancer.common.oracle.DQEBase;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLQueryError;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.common.schema.AbstractRelationalTable;
+import sqlancer.common.schema.AbstractTable;
+import sqlancer.mysql.MySQLErrors;
+import sqlancer.mysql.MySQLGlobalState;
+import sqlancer.mysql.MySQLSchema;
+import sqlancer.mysql.MySQLSchema.MySQLColumn;
+import sqlancer.mysql.MySQLSchema.MySQLTable;
+import sqlancer.mysql.MySQLSchema.MySQLTables;
+import sqlancer.mysql.MySQLVisitor;
+import sqlancer.mysql.ast.MySQLExpression;
+import sqlancer.mysql.gen.MySQLExpressionGenerator;
+
+public class MySQLDQEOracle extends DQEBase<MySQLGlobalState> implements TestOracle<MySQLGlobalState> {
+
+    private final MySQLSchema schema;
+
+    public MySQLDQEOracle(MySQLGlobalState state) {
+        super(state);
+        schema = state.getSchema();
+
+        MySQLErrors.addExpressionErrors(selectExpectedErrors);
+
+        MySQLErrors.addExpressionErrors(updateExpectedErrors);
+        updateExpectedErrors.add("cannot be null");
+        updateExpectedErrors.add("Duplicate entry");
+        updateExpectedErrors.add("The value specified for generated column");
+
+        MySQLErrors.addExpressionErrors(deleteExpectedErrors);
+        deleteExpectedErrors.add("a foreign key constraint fails");
+    }
+
+    public void check() throws SQLException {
```

Comment:
This method is quite long. Is it possible to break it apart into multiple smaller methods, or would that require passing around many parameters? In addition, we could try to make it easier for other dialects to implement DQE by specifying some methods in the superclass that should be implemented.